modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQProducer.java

```diff
@@ -36,147 +36,202 @@
-	 * But instead of this, finally block is used in storeMethod to close the channels
-	 */
-	public boolean cleanup() {
-		return store.cleanup(null, false);
-	}
-
-	public boolean isInitialized() {
-		return isInitialized;
-	}
-
-	public String getId() {
-		return idString;
-	}
+    private static final Log log = LogFactory.getLog(RabbitMQProducer.class.getName());
+
+    private static final int DEFAULT_PRIORITY = 0;
+    private Connection connection;
+    private RabbitMQStore store;
+    private String routingKey;
+    private String exchangeName;
+    private boolean isInitialized = false;
+    private String idString; // ID of the MessageProducer
+    private boolean publisherConfirmsEnabled;
+
+    /**
+     * The RabbitMQ producer
+     *
+     * @param store the {@link RabbitMQStore} object
+     */
+    public RabbitMQProducer(RabbitMQStore store) {
+        if (store == null) {
+            log.error("Cannot initialize producer: " + getId());
+            return;
+        }
+        this.store = store;
+        isInitialized = true;
+    }
+
+    /**
+     * Store the given message into the queue and return whether the operation success or not
+     *
+     * @param synCtx Message to be saved.
+     * @return {@code true} if storing of the message is successful, {@code false} otherwise.
+     */
+    @Override
+    public boolean storeMessage(MessageContext synCtx) {
+        if (synCtx == null) {
+            return false;
+        }
+        if (connection == null) {
+            log.error(getId() + " cannot proceed. RabbitMQ Connection is null. Ignored MessageId: " +
+                    synCtx.getMessageID());
+            return false;
+        }
+        boolean result = false;
+        try (Channel channel = connection.createChannel()) {
+            if (publisherConfirmsEnabled) {
+                channel.confirmSelect();
+            }
+            StorableMessage storableMessage = MessageConverter.toStorableMessage(synCtx);
+            final byte[] message = serializeMessage(storableMessage);
+            final AMQP.BasicProperties basicProperties = getBasicProperties(synCtx, storableMessage);
+            publishMessage(channel, exchangeName, routingKey, basicProperties, message);
+            if (publisherConfirmsEnabled) {
+                result = channel.waitForConfirms();
+            } else {
+                result = true;
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (TimeoutException | IOException e) {
+            String errorMsg = getId() + ". Ignored MessageId: " + synCtx.getMessageID() + ". " +
+                    "Could not store message to store [" + store.getName() + "]. " +
+                    "Error:" + e.getLocalizedMessage();
+            log.error(errorMsg, e);
+        }
+        if (log.isDebugEnabled()) {
+            log.debug(getId() + ". Stored MessageId: " + synCtx.getMessageID());
+        }
```

Comment:
I think this has to be inside the try block. The log message will be confusing after an error situation.

Reply:
Ack.